entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md

```diff
@@ -92,6 +96,127 @@ modelBuilder.Entity<Blog>()
     .IsRequired();
 ```
 
+<a name="cosmos-partition-key"></a>
+
+### Cosmos: Partition key is now added to the primary key
+
+[Tracking Issue #15289](https://github.com/aspnet/EntityFrameworkCore/issues/15289)
+
+**Old behavior**
+
+The partition key property was only added to the alternate key that includes `id`.
+
+**New behavior**
+
+The partition key property is now also added to the primary key by convention.
+
+**Why**
+
+This makes the model better aligned with Azure Cosmos DB semantics and improves the performance of `Find` and some queries.
+
+**Mitigations**
+
+To prevent the partition key property to be added to the primary key configure it in `OnModelCreating`.
+
+```cs
+modelBuilder.Entity<Blog>()
+    .HasKey(b => b.Id);
+```
+
+<a name="cosmos-id"></a>
+
+### Cosmos: `id` property renamed to `__id`
+
+[Tracking Issue #17751](https://github.com/aspnet/EntityFrameworkCore/issues/17751)
+
+**Old behavior**
+
+The shadow property mapped to the `id` JSON property was also named `id`.
+
+**New behavior**
+
+The shadow property created by convention is now named `__id`.
+
+**Why**
+
+This makes it less likely that the `id` property clashes with an existing property on the entity type.
+
+**Mitigations**
+
+To go back to the 3.x behavior configure the `id` property in `OnModelCreating`.
+
+```cs
+modelBuilder.Entity<Blog>()
+    .Property<string>("id")
+    .ToJsonProperty("id");
+```
+
+<a name="cosmos-byte"></a>
+
+### Cosmos: byte[] is now stored as a base64 string instead of a number array
+
+[Tracking Issue #17306](https://github.com/aspnet/EntityFrameworkCore/issues/17306)
+
+**Old behavior**
+
+Properties of type byte[] were stored as a number array.
+
+**New behavior**
+
+Properties of type byte[] are now stored as a base64 string.
+
+**Why**
+
+This aligns better with expectations and is the default behavior of the major JSON serialization libraries.
+
+**Mitigations**
+
+Existing data stored as number arrays will still be queried correctly, but currently there isn't a supported way to change back the insert behavior. If this is blocking your scenario please comment on [this issue](https://github.com/aspnet/EntityFrameworkCore/issues/17306)
+
+<a name="cosmos-byte"></a>
+
+### Cosmos: byte[] is now stored as a base64 string instead of a number array
```

Comment:
repeated section

Reply:
Weird there's no warning for repeated anchor